### PR TITLE
Add gosec to SAST analysis

### PR DESCRIPTION
**Pull request by WaqarAhmedVC**
# Description

Add gosec to perform Static Application Security Testing (SAST) Analysis. 

Fixes # (issue)

- Non existent SAST tool in the build pipeline

## Type of change

Please delete options that are not relevant.

- [x] New feature - Adding SAST tool in workflow (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [x] Adding a new tool in the workflow file. Tested to perform the required security checks without breaking 

**Test Configuration**:

* NA

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

**Comment by darrenvechain** [comment]
@WaqarAhmedVC we can add gosec to the linters in our lint config:

https://github.com/vechain/thor/blob/master/.golangci.yml#L13

**Comment by WaqarAhmedVC** [comment]
@darrenvechain Thanks for pointing out. I was searching for this info but couldn't find it.

File: cmd/thor/utils.go [hunk 1]
```diff
@@ -485,7 +485,7 @@ func startAPIServer(ctx *cli.Context, handler http.Handler, genesisID thor.Bytes
 	handler = handleXGenesisID(handler, genesisID)
 	handler = handleXThorestVersion(handler)
 	handler = requestBodyLimit(handler)
-	srv := &http.Server{Handler: handler}
+	srv := &http.Server{Handler: handler, ReadHeaderTimeout: time.Second, ReadTimeout: 5 * time.Second}
```
Comment (leszek-vechain) [inline cmd/thor/utils.go:488]: is this forced by gosec being on ?

Comment (otherview) [reply cmd/thor/utils.go:488]: Yep !

File: api/accounts/accounts_test.go [hunk 2]
```diff
@@ -527,7 +527,7 @@ func httpPost(t *testing.T, url string, body interface{}) ([]byte, int) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	res, err := http.Post(url, "application/x-www-form-urlencoded", bytes.NewReader(data))
+	res, err := http.Post(url, "application/x-www-form-urlencoded", bytes.NewReader(data)) // nolint:gosec
```
Comment (darrenvechain) [inline api/accounts/accounts_test.go:530]: Could optionally disable gosec for `*_test.go`

Comment (otherview) [reply api/accounts/accounts_test.go:530]: True, I think typically it's preferable to have the need to be aware of the why/how for disabling the linter in such particular cases.

**Review by darrenvechain — APPROVED** [review]

**Pull request merged**